### DISCUSSION TITLE
fast test and benchmark of new 1.67 versions vs 1.63

### DISCUSSION THREAD

**Discussion by nnyanpasu** [YellowRoseCx/koboldcpp-rocm#50] [General]
![image](https://github.com/YellowRoseCx/koboldcpp-rocm/assets/136471054/0f141d00-a498-46e8-97af-810eda557ddc)
rx7900xtx, only vram

**Comment by nnyanpasu** [comment]
img gen works well, x4 experts loaded too

**Comment by brknsoul** [comment]
Synopsis: Vulkan from the main fork still seems to be faster.


Benchmark
openhermes-2.5-mistral-7b.Q4_K_M
RX 6600 XT, 8GB

![image](https://github.com/YellowRoseCx/koboldcpp-rocm/assets/1969381/89e5cd9b-ddad-49f7-a913-c7d6e2a3876a)

**Reply by nnyanpasu** [reply]
what arguments do you use so that the bench saves the results of generating the token/s, etc.? by default, this is not saved in my file.

**Reply by brknsoul** [reply]
I just copypasted the results to a text file and asked Gemini to turn it into a spreadsheet.

**Reply by YellowRoseCx** [reply]
> Synopsis: Vulkan from the main fork still seems to be faster.
> 
> Benchmark openhermes-2.5-mistral-7b.Q4_K_M RX 6600 XT, 8GB
> 
> ![image](https://private-user-images.githubusercontent.com/1969381/337532972-89e5cd9b-ddad-49f7-a913-c7d6e2a3876a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgyNDQ4NjEsIm5iZiI6MTcxODI0NDU2MSwicGF0aCI6Ii8xOTY5MzgxLzMzNzUzMjk3Mi04OWU1Y2Q5Yi1kZGFkLTQ5ZjctYTkxMy1jN2Q2ZTJhMzg3NmEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MTNUMDIwOTIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2Y3ZDAzOTc0NmM3MWRkNGZhNDUzZmYzYTdmN2E4ZmUwODNiNDgzNDViMjUxOWU4NWQ3MWRiNGVhODg4NTAxNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.lxFRaK4oWjGZE2vIxsYEUrxQX2F0YQEQ4oKP6nPSEfU)

when you do comparisons like that, make sure you show the amount of tokens processed and the amount of tokens generated

**Reply by brknsoul** [reply]
It's the standard --benchmark

**Comment by nnyanpasu** [comment]
![image](https://github.com/YellowRoseCx/koboldcpp-rocm/assets/136471054/36f0c1c1-49c4-471b-bce1-13c02119b8a4)
vulkan test

**Comment by dya3506** [comment]
![Benchmark](https://github.com/YellowRoseCx/koboldcpp-rocm/assets/46310273/75f82211-01db-4aff-b4fb-d1c2d6ee4337)
on rocm4allV3, 6600xt with 8gb vram
offloading 49 layers to gpu works well on rocm but would get errors on vulkan, so I reduced to 40 layers

**Comment by cloak1505** [comment]
koboldcpp's built-in benchmark was a source of my stress. It makes ROCm look like complete ass.

When I'm actually using it, ROCm prompt processing is slower while generating faster. But for some reason it generates slower in the benchmark as well... What does the benchmark do??

![1 67 vulkan and rocm testing](https://github.com/YellowRoseCx/koboldcpp-rocm/assets/170299980/b628a47d-2f18-4d7f-9ab3-e0c65191ffa7)
I can fit 16k context on 8B IQ4_XS without dropping BLAS batch size.

I can see that depending on the use case, ROCm *is* worse. The benchmark is weirding me out though.
